jwst/pipeline/calwebb_spec3.py

```diff
@@ -72,6 +75,18 @@ def process(self, input):
         output_file = input_models.meta.asn_table.products[0].name
         self.output_file = output_file
 
+        # Split the inputs into science and background
+        science_data, background_data = split_container(input_models)

```

Comment:
This `split_container` function is memory intensive for a large association.  And it's necessary unless the background step is skipped.  I would move this splitting down to `if self.master_background.skip:`.

Reply:
Yeah, I was thinking about that too, noticing how many times the models are opened now when turning on debug mode. Another alternative approach is to just loop through the asn_table info looking for background members, and break as soon as you find one. That completely avoids all the opening of the models themselves.

Reply:
Turns out is is also file handle intensive.  For our full 20+20 MIRI MRS dataset, I run out of file handles.

Reply:
OK, so I guess that's a pretty good reason to make the change, right?